### PR TITLE
Fixing and Cleaning GUI & add new features

### PR DESCRIPTION
**Pull request by VannesteFelix**
- ADD: Creating new mechanism to add animation fct to UI and a way to describe them easily
- ADD: import animation function from different path & FIX deprecated import lib
- ADD: possibility to add animation from different python file
- ADD: possibility to generate a test scene with a specific phase to test animation sequence
- ADD: possibility to test specific animation pattern
- ADD: possibility to test before reduction
- ADD: possibility to choose which phase we want to do during reduction
- ADD: mechanism to save and load last cfg of GUI
- ADD: default value to dialogForm
- General cleaning & Fix

### CODE REVIEW

File: tools/test/test_component.py [hunk 1]
```diff
@@ -47,9 +60,17 @@ def createScene(rootNode):
...
+    #                 surfaceColor=[0.0, 0.0, 1, 0.5],
+    #                 surfaceMeshFileName=surfaceMeshFileNameSofia)
+
+
+    # # Animation parameters
+    # listObjToAnimate = [ObjToAnimate(reduced_test.getLinkPath()[2:]+'/nord',animation.shakingAnimations.defaultShaking,duration=-1,**{'incr': 5, 'incrPeriod': 10, 'rangeOfAction': 40}),
+    #                     ObjToAnimate(reduced_test.getLinkPath()[2:]+'/ouest',animation.shakingAnimations.defaultShaking,duration=-1,**{'incr': 5, 'incrPeriod': 10, 'rangeOfAction': 40}),
+    #                     ObjToAnimate(reduced_test.getLinkPath()[2:]+'/sud',animation.shakingAnimations.defaultShaking,duration=-1,**{'incr': 5, 'incrPeriod': 10, 'rangeOfAction': 40}),
+    #                     ObjToAnimate(reduced_test.getLinkPath()[2:]+'/est',animation.shakingAnimations.defaultShaking,duration=-1,**{'incr': 5, 'incrPeriod': 10, 'rangeOfAction': 40})]
+
```
Comment (alxbilger) [inline tools/test/test_component.py:76]: Is it useful to keep?

File: tools/test/test_component.py [hunk 2]
```diff
@@ -22,6 +22,19 @@ def createScene(rootNode):
                         dt=1,
                         gravity=[0.0,-9810,0.0])
 
+
+    # rootNode.VisualStyle.displayFlags="showForceFields"
+    # rootNode.addObject('FreeMotionAnimationLoop')
+    # rootNode.addObject('GenericConstraintSolver', tolerance="1e-6", maxIterations="1000")
+    # rootNode.addObject(AnimationManager(rootNode))
```
Comment (alxbilger) [inline tools/test/test_component.py:29]: Is it useful to keep?

File: python/mor/gui/widget/widgetUtility.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    btn.setIcon(icon)
+    btn.setFixedWidth(30)
+    btn.setStyleSheet("border:0px")
+    btn.setDisabled(True)
+
+    phaseWidget._title_frame._hlayout.addWidget(checkBox)
+    phaseWidget._title_frame._hlayout.addWidget(btn)
+
+    # checkBox.setParent(self.grpBox_Execution)
+    # btn.setParent(self.grpBox_Execution)
```
Comment (alxbilger) [inline python/mor/gui/widget/widgetUtility.py:78]: Is it useful to keep?

File: python/mor/gui/widget/widgetUtility.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    :param items:
+    :param phases:
+    :param phaseItem:
+    :return:
+    '''
+    checkBox , btn = phaseItem[index]
+    phase.blockSignals(True)
+
+    if checkBox.isChecked() == True:
+        # checkBox.setDisabled(False)
```
Comment (alxbilger) [inline python/mor/gui/widget/widgetUtility.py:30]: Is it useful to keep?

File: python/mor/gui/widget/genericDialogForm.py [hunk 5]
```diff
@@ -108,19 +115,21 @@ def load(self,data):
...
+            self.setCurrentValues()
+        # else:
+        #     for label,widget in self.row.items():
+        #         print(label,widget)
+        #         if dataType == bool:
+        #             widget.setCheckState(False)
+        #         else:
+        #             widget.setText('')
+        #
+        #     self.setCurrentValues()
```
Comment (alxbilger) [inline python/mor/gui/widget/genericDialogForm.py:127]: Is it useful to keep?

File: python/mor/gui/widget/animationTableWidget.py [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+            sizeForRow = 31 #self.sizeHintForRow(0)
+            width = self.width()
+            
+        if nbrRow > 5:
+            nbrRow = 5
+
+        # print(self.width())
+        size = QtCore.QSize(width,sizeForRow*nbrRow+sizeHeader)
+
+        # print(self.size(),size,sizeHeader)
```
Comment (alxbilger) [inline python/mor/gui/widget/animationTableWidget.py:223]:
```suggestion
```

File: python/mor/gui/widget/animationTableWidget.py [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+            sizeForRow = defaultHeight
+            width = defaultWidth
+        else:
+            sizeForRow = 31 #self.sizeHintForRow(0)
+            width = self.width()
+            
+        if nbrRow > 5:
+            nbrRow = 5
+
+        # print(self.width())
```
Comment (alxbilger) [inline python/mor/gui/widget/animationTableWidget.py:220]:
```suggestion
```

File: python/mor/gui/widget/animationTableWidget.py [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+    def _fromUtf8(s):
+        return s
+
+
+class AnimationTableWidget(QTableWidget):
+
+    def __init__(self,value,exp_path):
+
+        super(AnimationTableWidget, self).__init__(value)
+        # QTableWidget.__init__(self,exp_path)
```
Comment (alxbilger) [inline python/mor/gui/widget/animationTableWidget.py:37]:
```suggestion
```

File: python/mor/gui/utility.py [hunk 9]
```diff
@@ -209,3 +221,52 @@ def update_progress(progress):
...
+            newTxt = ''
+        else:
+            newTxt = '/'.join(newTxt[:-1])+'/'
+            if newTxt[-1] == '/' :
+                newTxt = newTxt[:-1]
+
+    lineEdit.setText(newTxt)
+
+    lineEdit.completer().setCompletionPrefix(newTxt)
+    # lineEdit.completer().complete()
```
Comment (alxbilger) [inline python/mor/gui/utility.py:239]:
```suggestion
```

**Review by alxbilger — CHANGES_REQUESTED** [review]

File: tools/test/test_component.py (hunk 2 above)
Comment (VannesteFelix) [reply tools/test/test_component.py:29]: same

File: python/mor/gui/widget/genericDialogForm.py (hunk 5 above)
Comment (VannesteFelix) [reply python/mor/gui/widget/genericDialogForm.py:127]: No we can remove it
```suggestion
```

File: python/mor/gui/widget/widgetUtility.py (hunk 4 above)
Comment (VannesteFelix) [reply python/mor/gui/widget/widgetUtility.py:30]:
```suggestion
```

File: python/mor/gui/widget/widgetUtility.py (hunk 3 above)
Comment (VannesteFelix) [reply python/mor/gui/widget/widgetUtility.py:78]:
```suggestion
```

File: python/mor/gui/widget/widgetUtility.py [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    '''
+    Utility fct to add a checkboxes and a reset button on every phase layout
+
+    :param phaseWidget:
+    :param grpBox_Execution:
+    :return:
+    '''
+
+    # CheckBox
+    checkBox = QtWidgets.QCheckBox()
```
Comment (VannesteFelix) [inline python/mor/gui/widget/widgetUtility.py:57]:
```suggestion
    checkBox = QtWidgets.QCheckBox(grpBox_Execution)
```

**Review by VannesteFelix — COMMENTED** [review]
I removed the different unnecessary comment, a lot more remain unfortunately. It's a bad habit from my part, I need to implement or use a msg_debug in my gui for debug in the future...

**Pull request merged**